libraries/AP_LQR_Control/AP_LQR_Control.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+/// @file    AP_LQR_Control.h
+/// @brief   LQR Control algorithm. This is a instance of an
+/// AP_Navigation class
+
+/*
+ * Written by Akshath Singhal 2018
+ *
+ */
+
+#include <AP_Math/AP_Math.h>
+#include <AP_AHRS/AP_AHRS.h>
+#include <AP_Param/AP_Param.h>
+#include <AP_Navigation/AP_Navigation.h>
+#include <AP_SpdHgtControl/AP_SpdHgtControl.h>
+
+class AP_LQR_Control : public AP_Navigation {
+public:
+    AP_LQR_Control(AP_AHRS &ahrs, const AP_SpdHgtControl *spdHgtControl)
+        : _ahrs(ahrs)
+        , _spdHgtControl(spdHgtControl)
+    {
+        AP_Param::setup_object_defaults(this, var_info);
+    }
+
+    /* Do not allow copies */
+    AP_LQR_Control(const AP_LQR_Control &other) = delete;
+    AP_LQR_Control &operator=(const AP_LQR_Control&) = delete;
+
+    /* see AP_Navigation.h for the definitions and units of these
```

Comment:
It would be better to add the units to the function names and variables, that way the code is self documented.